README.md

```diff
@@ -95,40 +112,95 @@ python config_generator.py --config config_templates/hard_hard.json \
+``` 
+## 6.Analyse CSA results
+To generate violin plots and analyse results, put all CSA results file in the same folder (here `csa_nnunet_vs_ivadomed`) and run:
+
+```
+python analyse_csa_nnunet_ivadomed.py -i-folder ~/duke/projects/ivadomed/contrast-agnostic-seg/csa_measures_pred/csa_nnunet_vs_ivadomed/ \
+                                      -include ivado_hard_GT ivado_avg_bin_no_crop ivado_soft_no_crop csa_nnunet_soft_avg_all_no_crop \
+```
+* `-i-folder`: Path to folder containing CSA results from models to analyse
+* `-include`: names of the folder names to include in the analysis (one model = one foler)
+
+### 6.1.Analyse CSA IVADOMED only
+To generate violin plots to compare 4 ivadomed models (original):
 
 ```
 python gen_charts.py --contrasts T1w T2w T2star rec-average_dwi \
        --predictions_folder ../duke/projects/ivadomed/contrast-agnostic-seg/csa_measures_pred/group8-9_combined-2022-12-21/ \
        --baseline_folder ../duke/projects/ivadomed/contrast-agnostic-seg/archive_derivatives_softsegs-seg/contrast-agnostic-centerofmass-preprocess-clean-all-2022-10-22\results_MTS_renamed
 ```
-## Run qc report on prediction masks
+  
+## 7.Run qc report on prediction masks from other datasets
 
+1. Got inside the `scripts` folder:
+~~~
+cd scripts
 ~~~
-sct_run_batch -path-data <PATH_DATA> -path-out <PATH-OUT> -script-args <PATH_PRED_MASK> -jobs 20 -script run_qc_prediction.sh
+2. Run bash script to generate QC report from prediction masks.
+**Note:** For nnUnet, ensure `prefix` in the script `compute_csa_nnunet.sh` according to the preffix in the prediction name and `contrast` with the image contrast.
 ~~~
+sct_run_batch -path-data <PATH_DATA> -path-out <PATH-OUT> -script-args <PATH_PRED_MASK> -jobs 20 -script run_qc_prediction_XXX.sh
+~~~
+* `-path-data`: Path to the original dataset used to run inferences.
+* `-path-output`: Path to the results folder to save QC report
+* `-script`: Script `run_qc_prediction_XXX` corresponding to the dataset.
+* `-script-args`: Path to prediction masks for the specific dataset
+  
+### 7.1.Example running QC on prediction masks from nnUnet from other datasets
+~~~
+sct_run_batch -jobs 20 -path-data ~/data_nvme_sebeda/datasets/dcm-zurich/ \
+                       -path-output ~/data_nvme_sebeda/qc_dcm_zurich_sag_nnUnet_2023-05-30 \
+                       -script run_qc_prediction_dcm_zurich_sag.sh \
+                       -script-args ~/duke/temp/muena/contrast-agnostic/pure-inference/Dataset725_dcmZurichSagittalRPI/test713_softAvg/
+~~~
+
+## 8.Active learning procedure
+To extend the training set to other contrasts and to pathologies, we applided the segmentation model to other datasets, manually corrected the segmentations and added them to the training set.
+
+Here is the detailed procedure:
+
+1. Run inference on other datasets for the selected models and QC report.
+2. Select ~20 interesting images per datasets (in QC).
+3. Correct the inference on the selected subject if needed.
```

Comment:
Will the inference be BIDS-compatible? If so, we can add info about `manual-correction`:
```suggestion
3. Correct the inference on the selected subjects if needed (you can use [`manual-correction`](https://github.com/spinalcordtoolbox/manual-correction) script).
```